File: solutions/python/resistor-color-duo/1/resistor_color_duo.py
def value(colors):
    if len(colors) > 2:
        colors = colors[:2]
    color_map = {
      "black": 0,
      "brown": 1,
      "red": 2,
      "orange": 3,
      "yellow": 4,
      "green": 5,
      "blue": 6,
      "violet": 7,
      "grey": 8,
      "white": 9
}

    return int("".join(str(color_map[color]) for color in colors))
        
        

File: solutions/javascript/resistor-color-duo/1/resistor-color-duo.js
//
// This is only a SKELETON file for the 'Resistor Color Duo' exercise. It's been provided as a
// convenience to get you started writing code faster.
//
const COLORS = [ "black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white" ];

export const decodedValue = (colors) => {
  if (colors.length < 2){
    const[first,second,...rest] = colors;
    colors = [first,second];
  }
  return COLORS.indexOf(colors[0]) * 10 + COLORS.indexOf(colors[1])
};
